client/src/components/Versions/VersionsTable/VersionsTable.js

```diff
@@ -39,65 +40,97 @@ const VersionsTable = ({
     versions,
     renderDeleteVersionButton,
     showDownloadCount,
+    appId,
 }) => {
     const getDownloadUrl = useCreateGetDownloadUrl()
+    const [changelogVisible, setChangelogVisible] = useState(false)
+    const showChangeLog = () => {
+        setChangelogVisible(!changelogVisible)
+    }
+
+    const onCloseChangelog = () => {
+        setChangelogVisible(false)
+    }
 
     return (
-        <Table className={styles.table}>
-            <TableHead>
-                <TableRowHead>
-                    <TableCellHead>Version</TableCellHead>
-                    <TableCellHead>Channel</TableCellHead>
-                    <TableCellHead>DHIS2 version compatibility</TableCellHead>
-                    <TableCellHead>Upload date</TableCellHead>
-                    {showDownloadCount && (
-                        <TableCellHead>Downloads</TableCellHead>
-                    )}
-                    <TableCellHead></TableCellHead>
-                </TableRowHead>
-            </TableHead>
-            <TableBody>
-                {versions.map((version) => (
-                    <TableRow key={version.id}>
-                        <TableCell>{version.version}</TableCell>
-                        <TableCell className={styles.channelNameCell}>
-                            {appChannelToDisplayName[version.channel]}
-                        </TableCell>
-                        <TableCell>
-                            {renderDhisVersionsCompatibility(
-                                version.minDhisVersion,
-                                version.maxDhisVersion
-                            )}
-                        </TableCell>
-                        <TableCell>
-                            <span title={new Date(version.createdAt)}>
-                                {new Date(
-                                    version.createdAt
-                                ).toLocaleDateString()}
-                            </span>
-                        </TableCell>
+        <>
+            {changelogVisible && (
+                <ChangeLogViewer
+                    appId={appId}
+                    modalShown={changelogVisible}
+                    onCloseChangelog={onCloseChangelog}
+                />
```

Comment:
⚠️ (minor) - this will *both* hide/show using the `modalShown` prop AND mount/unmount with the conditional render.  We should do one or the other, but I don't think we need to do both.